src/main/cc/wfa/virtual_people/common/field_filter/utils/message_filter_util.cc

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 The Cross-Media Measurement Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "wfa/virtual_people/common/field_filter/utils/message_filter_util.h"
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/str_cat.h"
+#include "google/protobuf/message.h"
+#include "src/main/proto/wfa/virtual_people/common/field_filter.pb.h"
+#include "wfa/measurement/common/macros.h"
+
+namespace wfa_virtual_people {
+
+absl::StatusOr<FieldFilterProto> ConvertMessageToFilter(
+    const google::protobuf::Message& message) {
+  FieldFilterProto filter;
+  filter.set_op(FieldFilterProto::AND);
+
+  const google::protobuf::Reflection* reflection = message.GetReflection();
+  std::vector<const google::protobuf::FieldDescriptor*> field_descriptors;
+  reflection->ListFields(message, &field_descriptors);
+
+  for (const google::protobuf::FieldDescriptor* field_descriptor :
+          field_descriptors) {
+    if (field_descriptor->is_repeated()) {
+      return absl::InvalidArgumentError(absl::StrCat(
+          "Repeated field in message when converting to FieldFilterProto: ",
+          message.DebugString()));
+    }
+    FieldFilterProto* sub_filter = filter.add_sub_filters();
+    sub_filter->set_name(field_descriptor->name());
+    switch (field_descriptor->cpp_type()) {
+      case google::protobuf::FieldDescriptor::CPPTYPE_INT32:
+        {
+          sub_filter->set_op(FieldFilterProto::EQUAL);
```

Comment:
Since most of these use the EQUAL type, what do you think about setting the op to `EQUAL` before the switch and then only overwriting it if it's different?

Reply:
Done.